serve.go

```diff
@@ -58,29 +58,59 @@ func serve(c Connector) error {
 	if c.NewSpecification == nil {
 		return errors.New("Connector.NewSpecification is a required field")
 	}
+
 	if c.NewSource == nil {
 		c.NewSource = func() Source { return nil }
 	}
 	if c.NewDestination == nil {
 		c.NewDestination = func() Destination { return nil }
 	}
 
+	cfg, err := getPluginConfig()
+	if err != nil {
+		return fmt.Errorf("failed to get plugin configuration: %w", err)
+	}
+
 	return server.Serve(
 		func() pconnector.SpecifierPlugin {
 			return NewSpecifierPlugin(c.NewSpecification(), c.NewSource(), c.NewDestination())
 		},
-		func() pconnector.SourcePlugin { return NewSourcePlugin(c.NewSource()) },
-		func() pconnector.DestinationPlugin { return NewDestinationPlugin(c.NewDestination()) },
+		func() pconnector.SourcePlugin { return NewSourcePlugin(c.NewSource(), cfg) },
+		func() pconnector.DestinationPlugin { return NewDestinationPlugin(c.NewDestination(), cfg) },
 	)
 }
 
-// getGRPCTargetEnv checks the environment variable provided by conduit-connector-protocol
-// and returns an error if it is not specified or empty.
-func getGRPCTargetEnv() (string, error) {
-	value := os.Getenv(pconduit.EnvConduitConnectorUtilitiesGRPCTarget)
-	if value == "" {
-		return "", fmt.Errorf("%q is not set. This indicates you are using an old version of Conduit."+
-			"Please, consider upgrading to at least v0.11.0. https://github.com/ConduitIO/conduit/releases/latest", pconduit.EnvConduitConnectorUtilitiesGRPCTarget)
+func getPluginConfig() (pconnector.PluginConfig, error) {
+	token := os.Getenv(pconduit.EnvConduitConnectorSchemaToken)
+	if token == "" {
+		return pconnector.PluginConfig{}, missingEnvError(pconduit.EnvConduitConnectorSchemaToken, "v0.11.0")
+	}
+
+	return pconnector.PluginConfig{
+		Token: token,
+		// TODO parse from env vars
+		ConnectorID: "",
+		LogLevel:    0,
+	}, nil
+}
+
+// connectorUtilsAddress returns the address and token to be used for the connector utilities service.
+// The values are fetched from environment variables provided by conduit-connector-protocol.
+// The function returns an error if the environment variables are not specified or empty.
+func connectorUtilsAddress() (string, error) {
+	target := os.Getenv(pconduit.EnvConduitConnectorUtilitiesGRPCTarget)
```

Comment:
WDYT about renaming this env. var to `EnvConduitConnectorUtilitiesAddress`? Also maybe dropping `Conduit` from the prefix (the package is `pconduit`)?

Reply:
We talked about this briefly with Raul, I can tell you the reasoning behind the name. First, the constant name corresponds to the value of the environment variable prefixed with `Env`, which is just capitalized and in snake case (`CONDUIT_CONNECTOR_UTILITIES_GRPC_TARGET`). As for the name itself:
- `CONDUIT` - prefix to prevent clashes with any other variables
- `CONNECTOR_UTILITIES` - the terminology we use for these gRPC services
- `GRPC_TARGET` - comes from [`grpc.NewClient`](https://pkg.go.dev/google.golang.org/grpc#NewClient) where the first parameter is named `target`

I know it's verbose, but I feel like it's important to be explicit and descriptive when it comes to environment variables.